File: springboot-react-frontend/src/service/CustomerService.js
import axios from 'axios'



const CUSTOMER_API_URL="http://localhost:8090/customer";
class CustomerService{
    getAllCustomers(){
      return axios.get(CUSTOMER_API_URL+'/Customers', 
        { crossdomain: true }
         );
       
    }
    getFilteredCustomers(countryCode,state){
        return axios.get(CUSTOMER_API_URL+'/FilteredCustomers/?countryCode='+countryCode+'&state='+state,{ crossdomain: true });
 
     }
}
export default new CustomerService()

File: src/main/java/com/jumia/SQLite/springboot/service/CustomerService.java
package com.jumia.SQLite.springboot.service;

import com.jumia.SQLite.springboot.entity.Customer;
import com.jumia.SQLite.springboot.repository.CustomerDao;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

import static com.jumia.SQLite.springboot.service.CustomerServiceUtility.getCustomersWithSameCountryCode;
import static com.jumia.SQLite.springboot.service.CustomerServiceUtility.getCustomersWithValidNotValidState;

@Service
public class CustomerService {

    @Autowired
    private CustomerDao customerDao;


    //getting all customers with valid and not valid state
    public List<Customer> getCustomers() {
        return customerDao.findAll();
    }

    //first we filter the number based on validity
    //second we filter all number that match the country code number regex
    public List<Customer> getFilteredCustomers(String countryCode, int state) {
        List<Customer> customersMatchTheState = getCustomersWithValidNotValidState(customerDao.findAll(), state);
        return getCustomersWithSameCountryCode(customersMatchTheState, countryCode);

    }


}
